test/e2e/framework/util.go

```diff
@@ -1179,53 +1179,6 @@ func AllNodesReady(c clientset.Interface, timeout time.Duration) error {
 	return nil
 }
 
-// RestartKubelet restarts kubelet on the given host.
-func RestartKubelet(host string) error {
-	// TODO: Make it work for all providers and distros.
-	supportedProviders := []string{"gce", "aws", "vsphere"}
-	if !ProviderIs(supportedProviders...) {
-		return fmt.Errorf("unsupported provider for RestartKubelet: %s, supported providers are: %v", TestContext.Provider, supportedProviders)
-	}
-	if ProviderIs("gce") && !NodeOSDistroIs("debian", "gci") {
-		return fmt.Errorf("unsupported node OS distro: %s", TestContext.NodeOSDistro)
-	}
-	var cmd string
-
-	if ProviderIs("gce") && NodeOSDistroIs("debian") {
-		cmd = "sudo /etc/init.d/kubelet restart"
-	} else if ProviderIs("vsphere") {
```

Comment:
The above lines(1184-1196) becomes unnecessary because this function is called at vsphere only.